code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -181,13 +181,15 @@ Due to keyboard shortcuts, the second one is not necessarily the remote eye's lo
 
 ///Places every slime in storage on target turf
 /obj/machinery/computer/camera_advanced/xenobio/proc/slime_place(turf/open/target_turf)
-	spit_out(stored_slimes, target_turf)
-	for(var/mob/living/basic/slime/stored_slime in stored_slimes)
-		stored_slime.forceMove(target_turf)
-		stored_slime.visible_message(span_notice("[stored_slime] is spat out!"))
-		stored_slimes -= stored_slime
-		REMOVE_TRAIT(stored_slime, TRAIT_STASIS, XENOBIO_CONSOLE_TRAIT)
-	xeno_hud.on_update_hud(LAZYLEN(stored_slimes), monkeys, max_slimes)
+	if(stored_slimes.len > 0)
+		if(stored_slimes.len == 1)
+			target_turf.visible_message(span_notice("The slime is warped in!"))
+		else
+			target_turf.visible_message(span_notice("[stored_slimes.len] slimes are warped in!"))
+		for(var/mob/living/basic/slime/stored_slime in stored_slimes)
+			stored_slime.forceMove(target_turf)
+			stored_slimes -= stored_slime
+			REMOVE_TRAIT(stored_slime, TRAIT_STASIS, XENOBIO_CONSOLE_TRAIT)
```

Comment:
use early returns here, it reduces indentation depth and makes the code a bit easier to read.
```suggestion
	if(stored_slimes.len <= 0)
		return
	if(stored_slimes.len == 1)
		target_turf.visible_message(span_notice("The slime is warped in!"))
	else
		target_turf.visible_message(span_notice("[stored_slimes.len] slimes are warped in!"))
	for(var/mob/living/basic/slime/stored_slime in stored_slimes)
		stored_slime.forceMove(target_turf)
		stored_slimes -= stored_slime
		REMOVE_TRAIT(stored_slime, TRAIT_STASIS, XENOBIO_CONSOLE_TRAIT)
```